20200922_Bank/src/de/telran/dao/Bank.java

```diff
@@ -0,0 +1,135 @@
+        }
+        accounts[index] = accounts[size - 1];
+        size--;
+        return true;
+    }
+
+    public void displayAccounts(){
+        for (int i = 0; i < size; i++) {
+            System.out.println(accounts[i]);
+            System.out.println("---------------");
+        }
+    }
+
+    public BankAccount[] getExistingAccounts(){
+        BankAccount[] allAccounts = new BankAccount[size];
+        for (int i = 0; i < size; i++) {
+            allAccounts[i] = accounts[i];
+        }
+        return allAccounts;
+    }
+
+    public double getBalanceOfAccount(){
+        double sum = 0;
+        for (int i = 0; i < size; i++) {
+            sum = sum + accounts[i].getBalance();
+        }
+        return sum;
+    }
+
+    public boolean transferMoney(int accountFrom, int accountTo, double sum){
+        int indexFrom = checkAccount(accountFrom);
+        if (indexFrom < 0){
+            System.out.println("Withdrawal account is incorrect, please check and try again");
+            return false;
+        }
+        int indexTo = checkAccount(accountTo);
+        if (indexTo < 0 ){
+            System.out.println("Recipient's account is incorrect, please check and try again\n");
+            return false;
+        }
+        if (!accounts[indexFrom].credit(sum)){
+            System.out.println("Transaction is not possible, your balance is too low\n");
+            return false;
+        }
+        accounts[indexTo].debit(sum);
+        System.out.println("Transaction is finished, see details below");
+        System.out.println("------------------------------------------");
+        System.out.println(sum + " Euro was transferred to " + accounts[indexTo].getClient().toStringName());
+        System.out.println("Your balance is " + accounts[indexFrom].getBalance());
+        System.out.println();
+        return true;
+    }
+
+    public void getCountClientsPerSex(Sex sex){
+        count = 0;
+/*        System.out.println("The following " + sex + " clients are found:");
+        System.out.println("---------------------------------------");*/
+        for (int i = 0; i < size; i++) {
+            if (accounts[i].getClient().getSex().equals(sex)) {
+                count++;
+                //System.out.println(count + ". " + accounts[i].getClient().toStringName() + "\n");
+            }
+        }
+        System.out.println("-------------------------------");
+        System.out.println("Total: " + count + " " + sex + " clients are found");
+        System.out.println("-------------------------------");
+    }
+
+    public Person[] getClientsPerSex(Sex sex){
+        int j = 0;
+        Person[] clients = new Person[count];
+        for (int i = 0; i < size; i++) {
+            if (accounts[i].getClient().getSex().equals(sex)) {
+                clients[j] = accounts[i].getClient();
+                j++;
+            }
+        }
+        return clients;
+    }
+
+    public void displayClientsPerSex(Person[] persons){
```

Comment:
Your method prints all the people from the array persons, not differentiate them by gender.